src/discovery/srv.go

```diff
@@ -57,52 +57,50 @@ func srvFetch(cfg config.DiscoveryConfig) (*[]core.Backend, error) {
 		return &[]core.Backend{}, nil
 	}
 
-	/* ----- try to get A data from additional section ------ */
+	/* ----- try to get IPs from additional section ------ */
 
 	hosts := make(map[string]string) // name -> host
 	for _, ans := range r.Extra {
-		record, ok := ans.(*dns.A)
-		if !ok {
-			continue
+		switch record := ans.(type) {
+		case *dns.A:
+			hosts[record.Header().Name] = record.A.String()
+		case *dns.AAAA:
+			hosts[record.Header().Name] = fmt.Sprintf("[%s]", record.AAAA.String())
```

Comment:
This is slightly annoying - Gobetween uses `net.Dial("tcp", addr)` which requires bracketed IPv6s. If storing IPs here, it would be better to store them as net.IPs.